pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -367,20 +346,36 @@ def _apply_lightning_controlled(self, sim, operation):
             Returns:
                 array[complex]: the output state tensor
             """
-            basename = "PauliX" if operation.name == "MultiControlledX" else operation.base.name
-            if basename == "Identity":
-                return
-            method = getattr(sim, f"{basename}", None)
             control_wires = self.wires.indices(operation.control_wires)
-            control_values = (
-                [bool(int(i)) for i in operation.hyperparameters["control_values"]]
-                if operation.name == "MultiControlledX"
-                else operation.control_values
-            )
+
             if operation.name == "MultiControlledX":
+                basename = "PauliX"
+                control_values = [bool(int(i)) for i in operation.hyperparameters["control_values"]]
                 target_wires = list(set(self.wires.indices(operation.wires)) - set(control_wires))
             else:
+                basename = operation.base.name
+                control_values = operation.control_values
                 target_wires = self.wires.indices(operation.target_wires)
+
+            if basename == "Identity":
+                return
+            # This special handling is required because PennyLane doesn't canonicalize
+            # C(CNOT/Toffoli) in qml.simplify or qml.ctrl/Controlled, and we now allow `C(...)`
+            # instances through for any supported base gate. However, the C++ simulator `method`
+            # for CNOT/Toffoli does not accept additional control wires.
+            elif basename == "CNOT":
+                basename = "PauliX"
+                control_values += [True]
+                control_wires += operation.base.wires[0]
+                target_wires = [operation.base.wires[1]]
+            elif basename == "Toffoli":
+                basename = "PauliX"
+                control_values += [True, True]
+                control_wires += operation.base.wires[0:2]
+                target_wires = [operation.base.wires[2]]
```

Comment:
I believe after my PR, if you call `operation.simplify()`, it would flatten any nested controlled structures to a multi-controlled structure. You can test it out and let me know if anything doesn't work. This would allow you to get rid of these special treatment.

Reply:
```
>>> import pennylane as qml
>>> from pennylane.ops import Controlled
>>> op = Controlled(qml.CNOT([0,1],2))
>>> op.simplify()
Toffoli(wires=[2, 0, 1])
```